File: labs_26_04_18/ex1/Hexagon.java
package labs_26_04_18.ex1;

// Klasa dla Sześciokąta foremnego
public class Hexagon extends Figure {
    private final double side;

    public Hexagon(double side) {
        if (side <= 0)
            throw new IllegalArgumentException("Bok musi być większy od zera.");
        this.side = side;
    }

    @Override
    public double calculateArea() {
        return (3 * Math.sqrt(3) * side * side) / 2.0;
    }

    @Override
    public double calculatePerimeter() {
        return 6 * side;
    }

    @Override
    public String getName() {
        return "Sześciokąt foremny";
    }
}

File: labs_26_04_18/ex2/src/figures/hexagon.cpp
#include "../../inc/hexagon.hpp"
#include <cmath>
#include <stdexcept>

Hexagon::Hexagon(double s) : side(s) {
  if (s <= 0)
    throw std::invalid_argument("Bok musi byc wiekszy od zera.");
}
double Hexagon::calculateArea() const {
  return (3 * std::sqrt(3) * side * side) / 2.0;
}
double Hexagon::calculatePerimeter() const { return 6 * side; }
std::string Hexagon::getName() const { return "Szesciokat foremny"; }